### PR TITLE
Add `parents` and `burned_mana` to rust block

### LINKED ISSUE #883: Add new parents field to Block - rust
https://github.com/iotaledger/tips-draft/blob/tip46/tips/TIP-0046/tip-0046.md#basic-block

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

This PR adds `WeakParents` and `ShallowLikeParents` to the block.

## Links to any relevant issues

Closes #883 
Closes https://github.com/iotaledger/iota-sdk/issues/886

### CODE REVIEW

File: sdk/src/types/block/core.rs [hunk 1]
```diff
@@ -12,6 +11,7 @@ use packable::{
     Packable, PackableExt,
 };
 
+use super::parent::{ShallowLikeParents, WeakParents};
```
Comment (thibault-martinez) [inline sdk/src/types/block/core.rs]: Put them together with the StrongParents import

File: sdk/src/types/block/parent.rs [hunk 2]
```diff
@@ -55,11 +54,21 @@ impl<const MIN: u8, const MAX: u8> Parents<MIN, MAX> {
...
+    }
+
     /// Returns the number of parents.
     pub fn len(&self) -> usize {
         self.0.len()
     }
 
+    /// Returns whether the parents list is empty.
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
```
Comment (Alex6323) [inline sdk/src/types/block/parent.rs:69]:
```suggestion
        self.0.is_empty()
```

Comment (DaughterOfMars) [reply sdk/src/types/block/parent.rs:69]: I'd rather use the type's impl, just because it's more flexible.

Comment (Alex6323) [reply sdk/src/types/block/parent.rs:69]: okay

**Review by Alex6323 — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by qrayven — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**